adplug/fprovide.cpp

```diff
@@ -20,6 +20,9 @@
  */
 
 #include <string.h>
+#ifdef GEKKO
+#include <strings.h>
```

Comment:
Better to have here include `pal_config.h` and include `strings.h` inside the configuration header.